docs/reference/extensions/go-framework.rst

```diff
@@ -0,0 +1,71 @@
+.. _go-framework-reference:
+
+go-framework
+----------------
+
+The Go extension streamlines the process of building Go application
+rocks.
+
+The extension builds and copies the Go binary file to the rock,
+inside the ``/app`` directory. By default, the base ``bare`` is used,
+to generate a lightweight image.
+
+
+.. note::
+    The Go extension is compatible with the ``bare`` and ``ubuntu@24.04``
+    bases.
+
+Project requirements
+====================
+
+To use the ``go-framework`` extension, there must be a ``go.mod`` file
+in the root directory of the project.
+
+
+``parts`` > ``go-framework/install-app`` > ``organize``
+=========================================================
+
+If the main package is in the base directory and the rockcraft name
+attribute is equal to the go module name, the name of the binary will
+be selected correctly, otherwise you can adjust it.
+
+You can use this field to specify a different binary to be used as the
+main application, without having to override the service command. For example,
+if your Go application contains a ``main`` package in the directory
+``cmd/anotherserver``, the name of the binary will be ``anotherserver``
+and you can override the main application to use the binary with the
+next snippet:
+
+.. code-block:: yaml
+
+  parts:
+    go-framework/install-app:
+     organize:
+       bin/anotherserver: usr/local/bin/<rockcraft project name>
```

Comment:
Would `bin/<rockcraft project name>` not work? :O

Reply:
`bin/<rockcraft project name>: usr/local/bin/<rockcraft project name>` is the "default one". 

If there are several binaries, and we want to include several of them in the rock, someway we have to choose the main one, and it will be the one in `usr/local/<rockcraft project name>`

Reply:
Resolving this conversation. Please, reopen it it you think it is still relevant.